datasciencebr/jarbas
# Delegate serialize to Celery on data import

**What is the purpose of this Pull Request?**
When trying to import data on the production server it **looks like** `rows` was unable to load and convert all data types for the whole 1.6 million lines of the source CSV. At this point the process just freezes after uncompressing the `.xz` file. This PR tries to fix this issue.

**What was done to achieve this purpose?**
The huge CSV is loaded line by line with native `csv.DictReader` and all fields are kept as raw/string format when passed to Celery. Then the conversion of data types happens line by line in the async/background process (not for the whole CSV as `rows` default behavior).

**How to test if it really works?**
Locally just run the command to import reimbursements and checks the database or the UI to make sure it worked. To have an idea if this is gonna work in the production server one might try to run it in a virtual machine with 4Gb of RAM (personally I haven't tested it, but once it's merged I can run the command in production and check it).

**Who can help reviewing it?**
@anaschwendler @irio
